src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -124,10 +124,16 @@ internal virtual async Task FlushLogs(CloudTable table = null)
                 return;
             }
 
-            table = table ?? GetDiagnosticEventsTable();
-
             try
             {
+                table = table ?? GetDiagnosticEventsTable();
```

Comment:
`table` would be null when AzureWebJobsStorage is not set.

Reply:
Possible to add check for AzureWebJobsStorage app setting - disable the feature - perhaps here - https://github.com/Azure/azure-functions-host/blob/de24ce6714a2092f05f2f63ff34f64beba880ce4/src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventLogger.cs#L37 or  here https://github.com/Azure/azure-functions-host/blob/de24ce6714a2092f05f2f63ff34f64beba880ce4/src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs#L148 ?

Reply:
The pattern for this repository is based on TableStorageScaleMetricsRepository, and checking that code for usages of [GetTable](https://github.com/Azure/azure-functions-host/blob/de24ce6714a2092f05f2f63ff34f64beba880ce4/src/WebJobs.Script.WebHost/Scale/TableStorageScaleMetricsRepository.cs#L81) most do proper null checks. However I do see a couple that don't. Please fix those as well.

Reply:
@mathewc I see the code you are referencing in TableStorageScaleMetricsRepository. Would like to send a separate PR for that change we can keep the changes isolated.